### PR TITLE
update(docs, layout): fix breakpoints, documentation, and gettingStarted

### LINKED ISSUE #4895: Examples on site are very old and wrong for newer version
The site has a lot of older examples on the landing page which are outdated. Can we please update them?

Site link : https://material.angularjs.org/latest/#/getting-started

### LINKED ISSUE #5576: Inconsistencies in the grid system
I've been having trouble with the grid system breakpoints and it seems that it's been used inconsistently through the code.

My understanding is that this is the correct usage:

```css
@media (max-width: $layout-breakpoint-sm - 1) { ... }
@media (min-width: $layout-breakpoint-sm) and (max-width: $layout-breakpoint-md - 1) { ... }
@media (min-width: $layout-breakpoint-md) and (max-width: $layout-breakpoint-lg - 1) { ... }
@media (min-width: $layout-breakpoint-lg) { ... }
```

But there are places in code where such code can be found:

```css
@media (max-width: $layout-breakpoint-sm) { ... }
@media (min-width: $layout-breakpoint-sm) and (max-width: $layout-breakpoint-md) { ... }
@media (min-width: $layout-breakpoint-md) and (max-width: $layout-breakpoint-lg) { ... }
@media (min-width: $layout-breakpoint-lg) { ... }
```

It is both indeterministic and, in my opinion, wrong. The particular example is from [bottom-sheet.scss](https://github.com/angular/material/blob/master/src/components/bottomSheet/bottom-sheet.scss), but it can be others as well (toolbar, toast, ...).

It would be really nice if this would be consolidated.

### LINKED ISSUE #5646: flex-order doesn't work as expected
flex-order doesn't support negative values, but according to documentation negative values should be supported

### LINKED ISSUE #6056: Adaptive UI breakpoints missing and docs out of date
In the change log for rc6 it's mentioned that the adaptive UI breakpoints have changed:

> **Breaking Changes**
> 
> Added breakpoints: ```xs```, ```gt-xs```, ```xl``` per Material Design spec. Breakpoints ```sm``` and ```gt-sm``` have changed.

First of all, the docs are still for the pre-rc6 breakpoints (```sm``` = 600px, ```md``` = 960px, ...). There's no mention of ```xs``` or ```xl``` anywhere.

Additionally, there doesn't seem to be CSS classes for ```hide-gt-xs```, ```show-gt-xs```, ```layout-padding-gt-xs```, ```layout-margin-gt-xs```. There might be more, but at least the first 2 are crucial for a successful transition from rc5 to rc6.

Also there, are still inconsistencies I've mentioned in #5576.

### PR DESCRIPTION
**Pull request by ThomasBurleson**
* fix mixins for `gt-lg` and `xl`
* update `hide` API / mediaQueries
* fix `hide` and `show` SCSS and component mediaQuery(s)
* fix `flex-order` with negative ordering; support -20 to 20
* update component custom mediaQueries to be consistent with ranges defined for Layout mediaQueries

Fixes #4895. Fixes #5646. Fixes #6056. Fixes #5576.

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
@jelbourn, @robertmesserle, @EladBezalel, @topherfangio - please review ASAP as I want to push to master this morning... so @naomiblack can review and word-smith.

File: docs/app/js/app.js [hunk 1]
```diff
@@ -248,23 +248,32 @@ function(SERVICES, COMPONENTS, DEMOS, PAGES, $location, $rootScope, $http, $wind
...
+      }
+      ,{
         name: 'Options',
         id: 'layoutOptions',
         url: 'layout/options'
-      },{
-        name: 'Tips & Tricks', // Possibly rename to Troubleshooting
+      }
+      ,{
+        name: 'Troubleshooting', // Possibly rename to Troubleshooting
```
Comment (topherfangio) [inline docs/app/js/app.js]: Since you've renamed, perhaps we don't need the comment :smile:

File: docs/app/partials/getting-started.tmpl.html [hunk 2]
```diff
@@ -1,38 +1,84 @@
 <div ng-controller="GuideCtrl" class="doc-content">
   <md-content>
-    <p><em>New to Angular.js? Before getting into Angular Material, it might be helpful to
-      <a href="https://egghead.io/articles/new-to-angularjs-start-learning-here" target="_blank"
-         title="Link opens in a new window">read about the framework</a>.</em></p>
+    <p><em>New to Angular.js? Before getting into Angular Material, it might be helpful to:</em>
+    <ul>
+      <li>watch the videos about <a href="https://egghead.io/articles/new-to-angularjs-start-learning-here" target="_blank"
+               title="Link opens in a new window">Angular.js framework</a></li>
```
Comment (topherfangio) [inline docs/app/partials/getting-started.tmpl.html]: Possibly reword: "watch the videos about the Angular.js framework".

Also, why is the title for both of these "Link opens in a new window"? Shouldn't it be related to the link? _(I realize this was not your change, just wondering if there is a reason this was done.)_

File: docs/app/partials/getting-started.tmpl.html [hunk 3]
```diff
@@ -1,38 +1,84 @@
...
+      </li>
+    </ul>
 
-    <h2>How do I start?</h2>
+    <p></p>
+    <h2>How do I start with Angular Material?</h2>
     <ul style="margin-bottom: 2em;">
       <li><a href="http://codepen.io/collection/AxKKgY/" target="_blank"
-             title="Link opens in a new window">Fork a Codepen</a></li>
+             title="Link opens in a new window">Visit CodePen Community' </a></li>
```
Comment (topherfangio) [inline docs/app/partials/getting-started.tmpl.html]: Extra single quote and space (`' `) at the end of the link.

File: docs/app/partials/getting-started.tmpl.html [hunk 4]
```diff
@@ -1,38 +1,84 @@
...
-             title="Link opens in a new window">Clone a Github Starter Project</a></li>
-    </ul>
+             title="Link opens in a new window">Use the Github Starter Project</a></li>
 
-    <h3>Including Angular Material and its dependencies</h3>
-    <ul style="margin-bottom: 2em;">
-      <li><a href="https://github.com/angular/material#bower">Using Bower</a></li>
-      <li><a href="https://github.com/angular/material#cdn">Using a CDN</a> (example below)</li>
+      <li style="font-weight: 400;color: #106CC8;">Use 'Edit on CodePen' button on any of our Demos<br/>
+        <img src="https://cloud.githubusercontent.com/assets/210413/11568997/ed86795a-99b4-11e5-898e-1da172be80da.png" style="width:75%;margin: 10px 30px 0 0">
```
Comment (topherfangio) [inline docs/app/partials/getting-started.tmpl.html]: LOVE this addition; hopefully it will help people when creating issues too...

File: docs/app/partials/getting-started.tmpl.html [hunk 5]
```diff
@@ -1,38 +1,84 @@
...
-            src='//codepen.io/team/AngularMaterial/embed/JYvaxJ/?height=412&theme-id=20485&default-tab=result'
-            frameborder='no' allowtransparency='true' allowfullscreen='true' style='width: 100%;'>
-      See the Pen <a href='http://codepen.io/team/AngularMaterial/pen/JYvaxJ/'>Angular Material Basic App</a>
-      by Angular Material (<a href='http://codepen.io/AngularMaterial'>@AngularMaterial</a>) on
-      <a href='http://codepen.io'>CodePen</a>.
+    <p></p>
+    <h3>Build a Material Application (blank shell)</h3>
+
+    Without installing <b>any</b> libraries locally, you can use Google's CDN to reference our online, cached Angular libraries.
+    Here is a CodePen Material applications that shows how to do that:
```
Comment (topherfangio) [inline docs/app/partials/getting-started.tmpl.html]: Remove `s` from `applications`

File: docs/app/partials/layout-container.tmpl.html [hunk 6]
```diff
@@ -1,28 +1,11 @@
...
-    </i>
-  </p>
-
-  <h3>Layout Attribute</h3>
-  <p>
-    Use the <code>layout</code> attribute on an element to arrange its children
+    Use the <code>layout</code> attribute on an container element to arrange specify the layout direction for its children:
     horizontally in a row (<code>layout="row"</code>) or vertically in a column (<code>layout="column"</code>).
-    Row layout is the default if you specify the <code>layout</code> attribute without a value.
+    Note that <code>row</code> the default layout direction if you specify the <code>layout</code> attribute without a value.
```
Comment (topherfangio) [inline docs/app/partials/layout-container.tmpl.html]: Typo, add `is`: "Note that `<code>row</code>` **is** the default layout direction if..."

File: docs/app/partials/getting-started.tmpl.html [hunk 7]
```diff
@@ -49,14 +95,18 @@
...
+      Angular Material v1.0 has a few known integration issues with the following libraries.
+      These libraries are not compatible for dual-use with Angular Material:
     </p>
     <ul style="margin-bottom: 2em;">
       <li>
-        <a href="https://docs.angularjs.org/api/ngTouch">ngTouchs</a>
-        - This module interferes with ngMaterial's click, tap, and swipe support for touch-enabled devices.
+        <a href="https://docs.angularjs.org/api/ngTouch">ngTouch</a>
+        - This Angular.js Touch module interferes with ngMaterial's click, tap, and swipe support for touch-enabled devices.
       </li>
```
Comment (EladBezalel) [inline docs/app/partials/getting-started.tmpl.html]: Maybe we should add ngUpgrade

File: docs/app/partials/getting-started.tmpl.html [hunk 8]
```diff
@@ -49,14 +95,18 @@
...
-      Angular Material v1.0 has a few known integration issues with the following libraries:
+      Angular Material v1.0 has a few known integration issues with the following libraries.
+      These libraries are not compatible for dual-use with Angular Material:
     </p>
     <ul style="margin-bottom: 2em;">
       <li>
-        <a href="https://docs.angularjs.org/api/ngTouch">ngTouchs</a>
-        - This module interferes with ngMaterial's click, tap, and swipe support for touch-enabled devices.
+        <a href="https://docs.angularjs.org/api/ngTouch">ngTouch</a>
+        - This Angular.js Touch module interferes with ngMaterial's click, tap, and swipe support for touch-enabled devices.
```
Comment (EladBezalel) [inline docs/app/partials/getting-started.tmpl.html]: Maybe we should add ngUpgrade

File: docs/app/partials/layout-alignment.tmpl.html [hunk 9]
```diff
@@ -4,47 +4,72 @@
...
+
   <p>
-   See below for more examples:
+   Below is an interactive demo that let's you explore the visual results of the different settings:
   </p>
 
   <section class="layout-panel-parent">
     <div ng-panel="layoutDemo">
-      <docs-demo demo-title='layout="{{layoutDemo.direction}}" layout-align="{{layoutAlign()}}"'
+      <docs-demo demo-title='layout="{{layoutDemo.direction}}" &nbsp; &nbsp; &nbsp; layout-align="{{layoutAlign()}}"'
```
Comment (EladBezalel) [inline docs/app/partials/layout-alignment.tmpl.html]: `&nbsp; &nbsp; &nbsp;` is this really necessary ?

File: docs/app/partials/layout-container.tmpl.html [hunk 10]
```diff
@@ -50,6 +35,111 @@
...
+      Note that <code>layout</code> only affects the flow direction for that container's <b>immediate</b> children.
+    </p>
+
+  <hr>
+
+  <br/>
+  <h3>Layouts and Responsive Breakpoints</h3>
+
+  <p>
+    As discussed in <a href="layout/introduction">Layout Introduction page</a> you can
```
Comment (topherfangio) [inline docs/app/partials/layout-container.tmpl.html]: Suggestion: Add `the` - "As discussed in **the** `<a href=...`"

or just remove the word "page" to make it a bit more sentence-like.

File: docs/app/partials/layout-children.tmpl.html [hunk 11]
```diff
@@ -142,50 +180,81 @@
...
+           <td>flex-order-gt-md</td>
+           <td>width &gt;= <b>1280</b>px</td>
+         </tr>
+         <tr>
+           <td>flex-order-lg</td>
+           <td><b>1280</b>px &lt;= width &lt; <b>1920</b>px</td>
+         </tr>
+         <tr>
+           <td>flex-order-gt-lg</td>
+           <td>width &gt;= <b>1920/b>px</td>
```
Comment (EladBezalel) [inline docs/app/partials/layout-children.tmpl.html]: You missed `<`

File: docs/app/partials/layout-container.tmpl.html [hunk 12]
```diff
@@ -50,6 +35,111 @@
...
+    make your layout change depending upon the device view size by using <b>breakpoint alias</b> suffices.
```
Comment (topherfangio) [inline docs/app/partials/layout-container.tmpl.html]: `suffices` to `suffixes`?

**Comment by EladBezalel** [comment]
LGTM, will review this locally later.

File: docs/app/partials/layout-introduction.tmpl.html [hunk 13]
```diff
@@ -0,0 +1,364 @@
...
+    Angular Material's Layout features are purposed as syntactic HTML/CSS sugar to allow developers
+    to easily and intuitively create responsive HTML layouts... built using modern
+    <a href="http://www.w3.org/TR/css3-flexbox/">flexbox</a> specifications and features. For developers, this
+    Layout API is simply a list of HTML attributes (aka directives) that can be used on HTML containers and elements.
+  </p>
+
+
+  <p>
+    Using <b> HTML Attributes</b> as the API provides an easy way to set a value (eg. <code>layout="row"</code>) and
+    helps with separation of concerns: Attributes define layout while and CSS classes assign styling.
```
Comment (topherfangio) [inline docs/app/partials/layout-introduction.tmpl.html]: Typo: "while and"

File: docs/app/css/style.css [hunk 14]
```diff
@@ -874,3 +874,12 @@ table.attributes  tr  td:last-child {
 table.md-api-table  tr  td:first-child {
   font-weight: bold;
 }
+
+
+.layout_note {
+  font-size: 0.9em;
+    margin: -5px 40px 0px 20px;
```
Comment (jelbourn) [inline docs/app/css/style.css]: Weird indentation

File: docs/app/partials/getting-started.tmpl.html [hunk 15]
```diff
@@ -1,38 +1,84 @@
 <div ng-controller="GuideCtrl" class="doc-content">
   <md-content>
-    <p><em>New to Angular.js? Before getting into Angular Material, it might be helpful to
-      <a href="https://egghead.io/articles/new-to-angularjs-start-learning-here" target="_blank"
-         title="Link opens in a new window">read about the framework</a>.</em></p>
+    <p><em>New to Angular.js? Before getting into Angular Material, it might be helpful to:</em>
```
Comment (jelbourn) [inline docs/app/partials/getting-started.tmpl.html]: Just "New to Angular"

File: docs/app/partials/layout-introduction.tmpl.html [hunk 16]
```diff
@@ -0,0 +1,364 @@
...
+
+  <br/><br/>
+  <hr>
+  <br/>
+
+  <h3>Under-the-Hood</h3>
+
+  <p>
+    Due to performance problems when using Attribute Selectors with <b>IE10</b> and <b>IE11</b> browsers, our attributes are
+    dynamically converted to class selectors at runtime. And the Layout CSS classNames and styles have been each been
```
Comment (topherfangio) [inline docs/app/partials/layout-introduction.tmpl.html]: "been each been" -> "each been"

File: docs/app/partials/layout-container.tmpl.html [hunk 17]
```diff
@@ -1,28 +1,11 @@
...
-    <i>
-      Note: Due to performance problems related to attribute selectors in IE11, our attributes are converted
-      to class selectors at runtime. You should continue to use layout attribute selectors in your code.
-    </i>
-  </p>
-
-  <h3>Layout Attribute</h3>
-  <p>
-    Use the <code>layout</code> attribute on an element to arrange its children
+    Use the <code>layout</code> attribute on an container element to arrange specify the layout direction for its children:
```
Comment (topherfangio) [inline docs/app/partials/layout-container.tmpl.html]: "on an container" -> "on a container"

File: docs/app/partials/getting-started.tmpl.html (hunk 2 above)
Comment (jelbourn) [inline docs/app/partials/getting-started.tmpl.html]: Agree with Topher.
* Make the entire sentence a hyperlink
* "Angular.js" > "Angular"
* Update `title` to say what's behind the link

File: docs/app/partials/layout-container.tmpl.html (hunk 17 above)
Comment (topherfangio) [inline docs/app/partials/layout-container.tmpl.html]: "to arrange specify" -> "to specify"

File: docs/app/partials/layout-container.tmpl.html [hunk 18]
```diff
@@ -50,6 +35,111 @@
...
+       <td>layout-gt-md</td>
+       <td>width &gt;= <b>1280</b>px</td>
+     </tr>
+     <tr>
+       <td>layout-lg</td>
+       <td><b>1280</b>px &lt;= width &lt; <b>1920</b>px</td>
+     </tr>
+     <tr>
+       <td>layout-gt-lg</td>
+       <td>width &gt;= <b>1920/b>px</td>
```
Comment (topherfangio) [inline docs/app/partials/layout-container.tmpl.html]: Missing `<` from closing `</b>` tag.

File: docs/app/partials/getting-started.tmpl.html (hunk 4 above)
Comment (jelbourn) [inline docs/app/partials/getting-started.tmpl.html]: We should be hosting this image, not githubusercontent.

File: docs/app/partials/layout-children.tmpl.html [hunk 19]
```diff
@@ -1,29 +1,32 @@
...
         </div>
       </div>
     </demo-file>
   </docs-demo>
+
   <p>
     Add the <code>flex</code> attribute to a layout's child element and the element will flex (grow or shrink) to fit
     the area unused by other elements. <code>flex</code> defines how the element will ajust its size with respect to its
-    parent container and the other other elements within the container.
+    <u>parent</u> container and the other other elements within the container.
```
Comment (topherfangio) [inline docs/app/partials/layout-children.tmpl.html]: duplicate `other`

File: docs/app/partials/getting-started.tmpl.html (hunk 5 above)
Comment (jelbourn) [inline docs/app/partials/getting-started.tmpl.html]: I would rework this whole section as 
"See this example application on CodePen that loads the Angular Material library from the Google CDN."

File: docs/app/partials/getting-started.tmpl.html [hunk 20]
```diff
@@ -1,38 +1,84 @@
...
     <br/>
 
+        <hr>
 
-    <h3>Include Optional Dependencies</h3>
+        <h3>Our CodePen Community</h3>
     <p>
-      Angular Material integrates with some additional libraries which you may optionally include:
+      You can also visit our <a href="http://codepen.io/team/AngularMaterial/" target="_blank"
+                     title="Link opens in a new window">CodePen Community</a> online to explore more Angular Material samples and ideas.
```
Comment (jelbourn) [inline docs/app/partials/getting-started.tmpl.html]: You can also visit our <a>...</a> to explore more samples and ideas.

File: docs/app/partials/getting-started.tmpl.html [hunk 21]
```diff
@@ -1,38 +1,84 @@
...
     </p>
+      <div layout-align="center" style="width: 100%">
+      <a href="http://codepen.io/team/AngularMaterial/" target="_blank"
+                     title="Link opens in a new window" style="text-decoration:none; border: 0 none;">
+      <img  src="https://cloud.githubusercontent.com/assets/210413/11569469/ada7de16-99b7-11e5-91b5-cf252a7d9122.png" style="text-decoration:none; border: 0 none;">
```
Comment (jelbourn) [inline docs/app/partials/getting-started.tmpl.html]: Same for this image.

File: docs/app/partials/layout-children.tmpl.html [hunk 22]
```diff
@@ -142,50 +180,81 @@
...
+
+  <h3>Add Offsets to the Preceding HTML Elements</h3>
+
+  <p>
+    Add the <code>flex-offset</code> attribute to a layout child to set its
+    offset percentage within the layout container. Values must be multiples
+    of <code>5</code> or <code>33</code> / <code>66</code>. These offsets establish a <code>margin-left</code>
+    with respect to the preceding element or the containers left boundary.
+  </p>
+
```
Comment (topherfangio) [inline docs/app/partials/layout-children.tmpl.html]: Might want to add a note here that this doesn't automatically switch to the right side if you use `flex-order` or manually set the direction to `flex-reverse`.

File: docs/app/partials/getting-started.tmpl.html (hunk 8 above)
Comment (jelbourn) [inline docs/app/partials/getting-started.tmpl.html]: No need to talk about ngUpgrade just yet.

File: docs/app/partials/getting-started.tmpl.html [hunk 23]
```diff
@@ -49,14 +95,18 @@
...
     </ul>
 
+    <br/>
+    <hr>
+
     <h3>Unsupported Integrations</h3>
     <p>
-      Angular Material v1.0 has a few known integration issues with the following libraries:
+      Angular Material v1.0 has a few known integration issues with the following libraries.
+      These libraries are not compatible for dual-use with Angular Material:
```
Comment (jelbourn) [inline docs/app/partials/getting-started.tmpl.html]: This whole paragraph can just be shorted to 
```
Angular Material v1.0 has a few known integration issues 
with the following libraries:
```

File: docs/app/partials/layout-alignment.tmpl.html [hunk 24]
```diff
@@ -4,47 +4,72 @@
     The <code>layout-align</code> attribute takes two words.
     The first word says how the children will be aligned in the layout's direction, and the second word says how the children will be aligned perpendicular to the layout's direction.</p>
 
-    <p>Only one word is required for the attribute. For example, <code>layout="row" layout-align="center"</code> would make the elements center horizontally and use the default behavior vertically.</p>
+    <p>Only one word is required for the attribute.
```
Comment (jelbourn) [inline docs/app/partials/layout-alignment.tmpl.html]: "one word" > "one value"

File: docs/app/partials/layout-alignment.tmpl.html [hunk 25]
```diff
@@ -4,47 +4,72 @@
...
+           <td>layout-align-xl</td>
+           <td>width &gt;= <b>1920</b>px</td>
+         </tr>
+        </table>
+
+  <br/>
+
   <p>
-   See below for more examples:
+   Below is an interactive demo that let's you explore the visual results of the different settings:
```
Comment (jelbourn) [inline docs/app/partials/layout-alignment.tmpl.html]: "let's" > "lets"

File: docs/app/partials/layout-options.tmpl.html [hunk 26]
```diff
@@ -22,38 +22,62 @@
...
     </tr>
     <tr>
       <td>layout-lg</td>
-      <td>Sets the layout on devices between 960 and 1200px wide (tablets in landscape).</td>
+      <td><b>1280</b>px &lt;= width &lt; <b>1920</b>px</td>
     </tr>
     <tr>
       <td>layout-gt-lg</td>
-      <td>Sets the layout on devices greater than 1200px wide (computers and large screens).</td>
+      <td>width &gt;= <b>1920/b>px</td>
```
Comment (topherfangio) [inline docs/app/partials/layout-options.tmpl.html]: Missing `<` on closing `</b>` tag.

File: docs/app/partials/layout-children.tmpl.html (hunk 19 above)
Comment (jelbourn) [inline docs/app/partials/layout-children.tmpl.html]: I recommend against using underline, since it is easily confused with a link.

File: docs/app/partials/layout-container.tmpl.html [hunk 27]
```diff
@@ -50,6 +35,111 @@
...
+   </table>
+   <br/>
+
+  <p><a
+      href="https://camo.githubusercontent.com/ad81ae92f8b4285747ce4e48007bf3f104dd5630/687474703a2f2f6d6174657269616c2d64657369676e2e73746f726167652e676f6f676c65617069732e636f6d2f7075626c6973682f6d6174657269616c5f765f342f6d6174657269616c5f6578745f7075626c6973682f3042386f6c5631354a3761625053474678656d46695156527462316b2f6c61796f75745f61646170746976655f627265616b706f696e74735f30312e706e67"
+      target="_blank" style="text-decoration: none;border: 0 none;">
+      <img
+      src="https://camo.githubusercontent.com/ad81ae92f8b4285747ce4e48007bf3f104dd5630/687474703a2f2f6d6174657269616c2d64657369676e2e73746f726167652e676f6f676c65617069732e636f6d2f7075626c6973682f6d6174657269616c5f765f342f6d6174657269616c5f6578745f7075626c6973682f3042386f6c5631354a3761625053474678656d46695156527462316b2f6c61796f75745f61646170746976655f627265616b706f696e74735f30312e706e67"
+      alt=""
+      style="max-width:100%;text-decoration: none;border: 0 none;"></a>
```
Comment (jelbourn) [inline docs/app/partials/layout-container.tmpl.html]: Same for these images.

File: docs/app/partials/layout-introduction.tmpl.html [hunk 28]
```diff
@@ -0,0 +1,364 @@
+<div ng-controller="LayoutCtrl" class="layout-content" ng-cloak>
+
+  <h3>Overview</h3>
+  <p>
+    Angular Material's Layout features are purposed as syntactic HTML/CSS sugar to allow developers
+    to easily and intuitively create responsive HTML layouts... built using modern
+    <a href="http://www.w3.org/TR/css3-flexbox/">flexbox</a> specifications and features. For developers, this
```
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: Prefer `https` links everywhere.

File: docs/app/partials/layout-children.tmpl.html (hunk 19 above)
Comment (topherfangio) [inline docs/app/partials/layout-children.tmpl.html]: Also, everywhere else that I saw you reference parent/child, you use bold, so I would stick with that.

File: docs/app/partials/layout-alignment.tmpl.html (hunk 9 above)
Comment (topherfangio) [inline docs/app/partials/layout-alignment.tmpl.html]: Not "necessary" per-se, but since this is the demo's title, it does help separate the the attributes a little bit and make them easier to read.

File: docs/app/partials/layout-introduction.tmpl.html [hunk 29]
```diff
@@ -0,0 +1,364 @@
+<div ng-controller="LayoutCtrl" class="layout-content" ng-cloak>
+
+  <h3>Overview</h3>
+  <p>
+    Angular Material's Layout features are purposed as syntactic HTML/CSS sugar to allow developers
```
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: "Angular Material's Layout features are purposed as syntactic HTML/CSS sugar to allow developers..."
>
"Angular Material's Layout features provide sugar to enable developers to more easily create modern, responsive layouts on top of CSS3 flexbox. The layout API consists of a set of Angular directives that can be applied to any of your application's HTML content."

File: docs/app/partials/layout-introduction.tmpl.html [hunk 30]
```diff
@@ -0,0 +1,364 @@
...
+    </tbody>
+  </table>
+
+
+  <p>And if we use Breakpoints as specified in Material Design:</p>
+  <p><a
+      href="https://camo.githubusercontent.com/ad81ae92f8b4285747ce4e48007bf3f104dd5630/687474703a2f2f6d6174657269616c2d64657369676e2e73746f726167652e676f6f676c65617069732e636f6d2f7075626c6973682f6d6174657269616c5f765f342f6d6174657269616c5f6578745f7075626c6973682f3042386f6c5631354a3761625053474678656d46695156527462316b2f6c61796f75745f61646170746976655f627265616b706f696e74735f30312e706e67"
+      target="_blank"><img
+      src="https://camo.githubusercontent.com/ad81ae92f8b4285747ce4e48007bf3f104dd5630/687474703a2f2f6d6174657269616c2d64657369676e2e73746f726167652e676f6f676c65617069732e636f6d2f7075626c6973682f6d6174657269616c5f765f342f6d6174657269616c5f6578745f7075626c6973682f3042386f6c5631354a3761625053474678656d46695156527462316b2f6c61796f75745f61646170746976655f627265616b706f696e74735f30312e706e67"
+      alt=""
```
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: Same for these images

**Comment by topherfangio** [comment]
@ThomasBurleson Finished a content review after reading through each of the layout docs. Are there any functionality changes you would like us to test as well?

File: docs/app/partials/layout-introduction.tmpl.html [hunk 31]
```diff
@@ -0,0 +1,364 @@
...
+    </tbody>
+  </table>
+
+  <br/>
+  <hr>
+  <h3>
+    API with Responsive Breakpoints
+  </h3>
+
+  <p>Now we can combine the breakpoint <code>alias</code> with the Layout API to easily support Responsive breakpoints
```
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: This section is worded like you're in the middle of a step-by-step tutorial rather than API docs. Instead something like
"The breakpoint alias combines with the Layout API, allowing..."

File: docs/app/partials/layout-introduction.tmpl.html [hunk 32]
```diff
@@ -0,0 +1,364 @@
...
+      </div>
+      <div flex class="three"></div>
+
+    </div>
+  </hljs>
+
+  <br/>
+
+  <p>
+  This Layout API means that developers no longer have to worry about the details of FlexBox styling.
```
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: "This Layout API means that developers no longer have to worry about the details of FlexBox styling." 

This is like saying GWT means that developers no longer have to worry about JavaScript. I would reword this to say that it makes it much easier to set up and maintain flexbox layouts vs. defining everything via CSS.

File: docs/app/partials/getting-started.tmpl.html (hunk 2 above)
Comment (ThomasBurleson) [inline docs/app/partials/getting-started.tmpl.html]: My interpretation is that when the user hovers over the link, it will show this tip of "intended action".

File: docs/app/partials/layout-alignment.tmpl.html (hunk 9 above)
Comment (ThomasBurleson) [inline docs/app/partials/layout-alignment.tmpl.html]: To separate them visually so users will see that *title* area shows the updated attribute values (when changed).

File: docs/app/partials/layout-container.tmpl.html (hunk 12 above)
Comment (ThomasBurleson) [inline docs/app/partials/layout-container.tmpl.html]: both are valid. latter is now more common.

File: docs/app/partials/layout-container.tmpl.html (hunk 27 above)
Comment (ThomasBurleson) [inline docs/app/partials/layout-container.tmpl.html]: ?? What do yo mean ?

File: docs/app/partials/layout-introduction.tmpl.html (hunk 31 above)
Comment (ThomasBurleson) [inline docs/app/partials/layout-introduction.tmpl.html]: True... the goal is full clarity of how the sugar works and why.

File: docs/app/partials/getting-started.tmpl.html (hunk 2 above)
Comment (jelbourn) [inline docs/app/partials/getting-started.tmpl.html]: >
The objective of this technique is to demonstrate how to use a title attribute on an anchor element to provide additional text describing a link. The title attribute is used to provide additional information to help clarify or further describe the purpose of a link. If the supplementary information provided through the title attribute is something the user should know before following the link, such as a warning, then it should be provided in the link text rather than in the title attribute.

http://www.w3.org/TR/WCAG20-TECHS/H33.html

File: docs/app/partials/layout-introduction.tmpl.html [hunk 33]
```diff
@@ -0,0 +1,364 @@
...
+  The developer uses the Layout HTML API to specify <b><i>intention</i></b> and the Layout engine handles all the issues of CSS flexbox styling.
+  </p>
+
+  <p class="layout_note">
+    The Layout engine will even report console warnings for Layout conflicts and issues that are not automatically resolved.
```
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: "will even" > "will"

File: docs/app/partials/layout-introduction.tmpl.html [hunk 34]
```diff
@@ -0,0 +1,364 @@
...
+
+
+  <br/><br/>
+  <hr>
+  <br/>
+
+  <h3>Under-the-Hood</h3>
+
+  <p>
+    Due to performance problems when using Attribute Selectors with <b>IE10</b> and <b>IE11</b> browsers, our attributes are
```
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: Just say "Internet Explorer" instead of IE10 and IE11. It would be good if we could provide a link with more information about said performance problems.

File: docs/app/partials/layout-introduction.tmpl.html (hunk 16 above)
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: We should mention that the CSS classes used by layout are considered private implementation details and may change between releases.

File: docs/app/partials/layout-introduction.tmpl.html (hunk 29 above)
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: For the rest of this section, I would use the term *"layout directive"* vs. "layout attribute". I think this makes it much more clear that we have code that's actually doing stuff, vs. just targeting the attributes with CSS.

File: docs/app/partials/layout-introduction.tmpl.html [hunk 35]
```diff
@@ -0,0 +1,364 @@
...
+    dynamically converted to class selectors at runtime. And the Layout CSS classNames and styles have been each been
+    predefined in the <code>angular-material.css</code> stylesheet.
+  </p>
+
+  <p class="layout_note">
+    This means the HTML design-time markup is not affected. Developers should continue to use Layout attribute selectors in the HTML
```
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: I think this paragraph is confusing. I don't know what "design-time" means or what it's not affected by. "Developers should continue to use Layout attribute": as opposed to what?

I would replace this paragraph with a note that the Layout directives do more than directly apply CSS classes, and that the classes may change between releases. Using the directives is the only supported API.

File: docs/app/partials/layout-introduction.tmpl.html [hunk 36]
```diff
@@ -0,0 +1,364 @@
...
+        <div flex="33">Second item in column</div>
+
+      </div>
+
+    </div>
+  </hljs>
+
+
+  <p>
+    At runtime, these attributes are transformed to CSS classnames.
```
Comment (jelbourn) [inline docs/app/partials/layout-introduction.tmpl.html]: "...to CSS classes."

File: docs/app/partials/layout-options.tmpl.html [hunk 37]
```diff
@@ -97,77 +122,36 @@
...
-    </tr>
-    <tr>
-      <td>flex-gt-lg</td>
-      <td>Sets flex on devices greater than 1200px wide.</td>
-    </tr>
-  </table>
+    <h3>Show &amp; Hide </h3>
 
-  <br/>
+  <p>Use the <code>show</code> <code>hide</code> APIs to responsively show or hide elements</p>
```
Comment (jelbourn) [inline docs/app/partials/layout-options.tmpl.html]: Might want to mention that these work like `ng-show` and `ng-hide` (vs. `ng-if`)
